Fix sitemap dumper

alexander-schranz opened the pull request:
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | yes
| Deprecations? | no
| Fixed tickets | fixes #4094 fixes #4327
| Related issues/PRs | -
| License | MIT
| Documentation PR | https://github.com/sulu/sulu-docs/pull/456

#### What's in this PR?

Fix sitemap dumper by going on host and not on a single portal or language.

#### Why?

Currently the sitemap dumper does not work as it will dump only for once portal and one locale. For this several interfaces in the sitemap namespace need to be changed to only have 

#### To Do

- [x] Documentation
- [x] Add breaking changes to UPGRADE.md

danrot commented on src/Sulu/Component/Webspace/Manager/WebspaceManager.php:
```diff
@@ -98,6 +98,34 @@ public function findPortalInformationByUrl(string $url, ?string $environment = n
 
+    public function findPortalInformationsByHost(
+        string $host,
+        bool $includeSubdomains = false,
```
Is that parameter needed? I have never seen this function being called with a different value than `true`...

danrot replied:
Then I would rename the function, instead of introducing a basically unused parameter. How about `findPortalInformationsByHostIncludingSubdomains`? Would also make the calling code easier to understand, otherwise you always have to lookup what this boolean flag means...

alexander-schranz commented:
@danrot I moved the service tag alias to a `getAlias` function on the SitemapProviderInterface, so its now autowireable. Docs are also updated: https://github.com/sulu/sulu-docs/pull/456

alexander-schranz commented on src/Sulu/Bundle/WebsiteBundle/Sitemap/Provider/PagesSitemapProvider.php:
```diff
@@ -35,75 +38,136 @@ class PagesSitemapProvider implements SitemapProviderInterface
+        string $scheme
+    ) {
+        $changed = $contentPage['changed'];
+        if (is_string($changed)) {
```
should only be string here. Seems like new \DateTime is only coming from the UnitTests

danrot commented on UPGRADE.md:
```diff
@@ -1,5 +1,65 @@
+
+As a sitemap is always domain specific and a domain can have multiple webspaces and portal
+the `SitemapProviderInterface` changed. To make it possible to provide also none Sulu
+routes the `SitemapUrl` constructor changed and now needs as third paramter a defaultLocale.
```
Use backticks for `defaultLocale` and you've got a typo in "parameter". I would also change the sentence to "constructor introduced a `defaultLocale` as third parameter"

danrot commented on UPGRADE.md:
````diff
@@ -1,5 +1,65 @@
+}
+```
+
+In this case also the `XmlSitemapDumper` and `XmlSitemapRender` does not longer need PortalInformations.
````
"In this case" does not really fit here... How about "The `XmlSitemapDumper` and `XmlSitemapRenderer` no longer need `PortalInformations`."
